certs/api/transport.go

```diff
@@ -0,0 +1,172 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/mainflux/mainflux/certs"
+	"github.com/mainflux/mainflux/pkg/errors"
+
+	kithttp "github.com/go-kit/kit/transport/http"
+	"github.com/go-zoo/bone"
+	"github.com/mainflux/mainflux"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	contentType = "application/json"
+	offset      = "offset"
+	limit       = "limit"
+
+	defOffset = 0
+	defLimit  = 10
+)
+
+var (
+	errUnsupportedContentType = errors.New("unsupported content type")
+	errUnauthorized           = errors.New("missing or invalid credentials provided")
+	errInvalidQueryParams     = errors.New("invalid query params")
+	errMalformedEntity        = errors.New("malformed entity")
+	errConflict               = errors.New("entity already exists")
+)
+
+// MakeHandler returns a HTTP handler for API endpoints.
+func MakeHandler(svc certs.Service) http.Handler {
+
+	opts := []kithttp.ServerOption{
+		kithttp.ServerErrorEncoder(encodeError),
+	}
+
+	r := bone.New()
+
+	r.Post("/certs", kithttp.NewServer(
+		issueCert(svc),
+		decodeCerts,
+		encodeResponse,
+		opts...,
+	))
+
+	r.Get("/certs", kithttp.NewServer(
+		listCerts(svc),
+		decodeListCerts,
+		encodeResponse,
+		opts...,
+	))
+
+	r.Post("/certs/revoke", kithttp.NewServer(
```

Comment:
Is DELETE more suitable for revocation?

Reply:
i made it `POST` because in vault revoke is `POST`